cpp/src/datetime/datetime_ops.cu

```diff
@@ -442,6 +534,50 @@ gdf_error gdf_extract_datetime_day(gdf_column *input, gdf_column *output) {
 	return GDF_SUCCESS;
 }
 
+gdf_error gdf_extract_datetime_weekday(gdf_column *input, gdf_column *output) {
+
+	GDF_REQUIRE(input->size == output->size, GDF_COLUMN_SIZE_MISMATCH);
+	GDF_REQUIRE(output->dtype == GDF_INT16, GDF_UNSUPPORTED_DTYPE);
+
+	cudaStream_t stream;
+	cudaStreamCreate(&stream);
+
+
+    if (input->valid){
+      gdf_size_type num_bitmask_elements = ( ( input->size +( GDF_VALID_BITSIZE - 1)) / GDF_VALID_BITSIZE );
+      thrust::copy(rmm::exec_policy(stream)->on(stream), input->valid, input->valid + num_bitmask_elements, output->valid); // copy over valid bitmask
+    }
+	if ( input->dtype == GDF_DATE64 ) {
```

Comment:
This needs to use the [`type_dispatcher`](https://github.com/rapidsai/cudf/blob/branch-0.10/cpp/include/cudf/utilities/legacy/type_dispatcher.hpp) with a specialization for the timestamp/date types.

Reply:
You can make `gdf_extract_weekday_from_unixtime_op` be a type-dispatched function object.